api/mapping/views.py

```diff
@@ -222,6 +221,13 @@ def get_serializer_class(self):
             return ProjectSerializer
         return ProjectNameSerializer
 
+    def get_queryset(self):
+        if self.request.GET.get("dataset") != None:
+            dataset = self.request.GET.get("dataset")
+            return Project.objects.filter(datasets__exact=dataset).distinct()
```

Comment:
```suggestion
        if dataset := self.request.GET.get("dataset"):
            return Project.objects.filter(datasets__exact=dataset).distinct()
```

Reply:
Not sure, but I suspect you can just do `...filter(datasets=dataset).distinct()`